src/Sulu/Bundle/AdminBundle/Resources/js/components/BlockCollection/SortableBlockList.js

```diff
@@ -66,7 +67,7 @@ class SortableBlockList extends React.Component<Props> {
                         activeType={block.type}
                         expanded={!disabled && expandedBlocks[index]}
                         index={index}
-                        key={block.__id}
+                        key={generatedBlockIds[index]}
```

Comment:
i am curious - does this have any other effect than improved performance?

Reply:
Not even sure if it has better performance :see_no_evil: But yes, it has another impact. This way the generated ID, which is only necessary because we need a `key` for react, was previously sent to the server. And if the API stored this ID somehow on the serverside and returned it when loading the entity, it lead to the weird behavior described in #5185 (adding a new block when some of the blocks were reordered).

Reply:
yeah i know about the weird behaviour 🙂 i mean using a generated id for the`key` prop instead of just using the index of the array element?

Reply:
If you could use the `index` of the current element, then the `key` property in react would not make any sense at all :wink: 

If you use the `index` in that case sorting the elements does not work correctly.